app/assets/javascripts/discourse/app/components/ace-editor.js

```diff
@@ -143,7 +147,7 @@ export default class AceEditor extends Component {
             editor.$blockScrolling = Infinity;
             editor.renderer.setScrollMargin(10, 10);
 
-            this.element.setAttribute("data-editor", editor);
+            this.element.dataset.editor = editor.id;
```

Comment:
This is fixing a mistake in core currently where we set a whole object on a data attribute